src/App.tsx

```diff
@@ -15,48 +17,89 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'name',
+  length = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  isRev: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortField) {
+      case SortType.length:
+        return good1.length - good2.length;
+
+      case SortType.name:
+        return good1.localeCompare(good2);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isRev) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [isRev, setIsReversed] = useState(false);
+
+  const preparedGoods = getPreparedGoods(goodsFromServer, sortField, isRev);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
+          onClick={() => setSortField(SortType.name)}
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortField !== SortType.name && 'is-light'}`}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={`button is-success ${sortField !== SortType.length && 'is-light'}`}
+          onClick={() => setSortField(SortType.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button is-warning', { 'is-light': !isRev })}
+          onClick={() => setIsReversed(!isRev)}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || isRev) && (
+          <button
+            onClick={() => {
+              setSortField('');
+              setIsReversed(false);
+            }}
```

Comment:
Move to separate function  (transfer the function to a variable, and here only write the name of the function)